Build fails if eslint or "stylish" eslint reporter are missing

If eslint is missing, the build fails; but if the specific "stylish" eslint reporter is missing, the build also fails.
```
    ERROR in ./themes/myproject/src/frontend.js
    Module build failed (from ./node_modules/eslint-loader/index.js):
    Error: Cannot find module 'eslint/lib/formatters/stylish'
        at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
        at Function.Module._load (internal/modules/cjs/loader.js:508:25)
        at Module.require (internal/modules/cjs/loader.js:637:17)
        at require (/hm/tmp/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
        at Object.module.exports (/hm/tmp/node_modules/eslint-loader/index.js:199:26)
```
This is not ideal because the build should work regardless of the specific eslint configuration in play. Either stylish should be bundled, or we should detect when it's not and not error.

--- a/src/helpers/is-installed.js
+++ b/src/helpers/is-installed.js
@@ -1,0 +1,16 @@
+/**
+ * Check whether a package may be `require`d without error.
+ *
+ * @param {String} packageName The package to test.
+ * @returns {Boolean} Whether the package is available via `require()`.
+ */
+module.exports = ( packageName ) => {
+	/* eslint-disable global-require */
+	try {
+		require( packageName );
+		return true;
+	} catch ( err ) {
+		return false;
+	}
+	/* eslint-enable */
+};

--- a/src/helpers/is-installed.test.js
+++ b/src/helpers/is-installed.test.js
@@ -1,0 +1,16 @@
+const isInstalled = require( './is-installed' );
+
+describe( 'isInstalled', () => {
+
+	it( 'is a function', () => {
+		expect( isInstalled ).toBeInstanceOf( Function );
+	} );
+
+	it( 'returns true for a package which is installed', () => {
+		expect( isInstalled( 'jest' ) ).toBe( true );
+	} );
+
+	it( 'returns false if the value is not found', () => {
+		expect( isInstalled( 'wpapi' ) ).toBe( false );
+	} );
+} );

--- a/src/presets.js
+++ b/src/presets.js
@@ -1,6 +1,7 @@
 const { devServer, stats } = require( './config' );
 const deepMerge = require( './helpers/deep-merge' );
 const findInObject = require( './helpers/find-in-object' );
+const isInstalled = require( './helpers/is-installed' );
 const loaders = require( './loaders' );
 const plugins = require( './plugins' );
 const { ManifestPlugin, MiniCssExtractPlugin } = plugins.constructors;
@@ -46,8 +47,8 @@ const development = ( options = {} ) => {
 		module: {
 			strictExportPresence: true,
 			rules: [
-				// Run all JS files through ESLint.
-				loaders.eslint(),
+				// Run all JS files through ESLint, if installed.
+				...( isInstalled( 'eslint' ) ? [ loaders.eslint() ] : [] ),
 				{
 					// "oneOf" will traverse all following loaders until one will
 					// match the requirements. If no loader matches, it will fall
@@ -168,8 +169,8 @@ const production = ( options = {} ) => {
 		module: {
 			strictExportPresence: true,
 			rules: [
-				// Run all JS files through ESLint.
-				loaders.eslint(),
+				// Run all JS files through ESLint, if installed.
+				...( isInstalled( 'eslint' ) ? [ loaders.eslint() ] : [] ),
 				{
 					// "oneOf" will traverse all following loaders until one will
 					// match the requirements. If no loader matches, it will fall

